kingpin/actors/group.py

```diff
@@ -338,14 +349,18 @@ def _run_actions(self):
         failed (False).
         """
 
-        # This is an interesting tornado-ism. Here we generate and fire off
-        # each of the acts asynchronously into the IOLoop, and we record
-        # references to those tasks. However, we don't yield (wait) on them to
-        # finish.
         tasks = []
         for act in self._actions:
             tasks.append(act.execute())
 
+            if self.option('concurrency') > 0:
+                running_tasks = len([t for t in tasks if t.running()])
+
+                while running_tasks == self.option('concurrency'):
+                    self.log.debug('Concurrecnty saturated. Waiting.')
+                    yield gen.sleep(0)
```

Comment:
Ideally I'd have some `noop()` function. I don't actually need to sleep, I just need to allow the tornado stack to progress due to the internal ioloop. Basically making this equivalent to "do *something* then check again"